service/resource/cloudformationv2/update_test.go

```diff
@@ -0,0 +1,134 @@
+				AZ: "myaz",
+				Masters: []v1alpha1.AWSConfigSpecAWSNode{
+					v1alpha1.AWSConfigSpecAWSNode{
+						ImageID: "myimageid",
+					},
+				},
+				Workers: []v1alpha1.AWSConfigSpecAWSNode{
+					v1alpha1.AWSConfigSpecAWSNode{
+						ImageID: "myimageid",
+					},
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		obj            interface{}
+		currentState   interface{}
+		desiredState   interface{}
+		expectedChange awscloudformation.UpdateStackInput
+		description    string
+	}{
+		{
+			description:  "current and desired state empty, expected empty",
+			obj:          clusterTpo,
+			currentState: StackState{},
+			desiredState: StackState{},
+			expectedChange: awscloudformation.UpdateStackInput{
+				StackName:  aws.String(""),
+				Parameters: []*awscloudformation.Parameter{},
+			},
+		},
+		{
+			description:  "current state empty, desired state not empty, expected desired state",
+			obj:          clusterTpo,
+			currentState: StackState{},
+			desiredState: StackState{
+				Name:           "desired",
+				Workers:        "4",
+				ImageID:        "ami-1234",
+				ClusterVersion: "myclusterversion",
+			},
+			expectedChange: awscloudformation.UpdateStackInput{
+				StackName: aws.String("desired"),
+			},
+		},
+		{
+			description: "current state not empty, desired state not empty but different, expected desired state",
+			obj:         clusterTpo,
+			currentState: StackState{
+				Name:           "current",
+				Workers:        "3",
+				ImageID:        "ami-6789",
+				ClusterVersion: "oldclusterversion",
+			},
+			desiredState: StackState{
+				Name:           "desired",
+				Workers:        "4",
+				ImageID:        "ami-1234",
+				ClusterVersion: "myclusterversion",
+			},
+			expectedChange: awscloudformation.UpdateStackInput{
+				StackName: aws.String("desired"),
+			},
+		},
+	}
+
+	var err error
+	var newResource *Resource
+	{
+		resourceConfig := DefaultConfig()
+		resourceConfig.Clients = &adapter.Clients{
+			EC2: &adapter.EC2ClientMock{},
+			IAM: &adapter.IAMClientMock{},
+			KMS: &adapter.KMSClientMock{},
+		}
+		resourceConfig.HostClients = &adapter.Clients{
+			IAM: &adapter.IAMClientMock{},
+			EC2: &adapter.EC2ClientMock{},
+		}
+		resourceConfig.Logger = microloggertest.New()
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Error("expected", nil, "got", err)
+		}
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			result, err := newResource.newUpdateChange(context.TODO(), tc.obj, tc.currentState, tc.desiredState)
+			if err != nil {
+				t.Errorf("expected '%v' got '%#v'", nil, err)
+			}
+			updateChange, ok := result.(awscloudformation.UpdateStackInput)
+			if !ok {
+				t.Errorf("expected '%T', got '%T'", updateChange, result)
+			}
+			/*
+				if updateChange.StackName != nil && *updateChange.StackName != *tc.expectedChange.StackName {
+					t.Errorf("expected %v, got %v", *tc.expectedChange.StackName, *updateChange.StackName)
+				}
+			*/
```

Comment:
Should this be commented out? If so can you add a comment when it can be enabled

Reply:
Sorry, i removed this while doing a huge rebase, the test was failing but looks to me like the test case was wrong, if current is empty and desired is not, the update change should be empty (the create change should not), could you please have another look?

Reply:
No worries and yes I agree the test is wrong. This case should be a create change.